### DISCUSSION TITLE
Buttons in the notification

### DISCUSSION THREAD

**Discussion by sarmong** [phuhl/linux_notification_center#197] [Q&A]
I have some question in regards to button in the notification itself.

1. Are they assigned by the app that sends that notification?
2. How can I style them?
3. How can I find out what they supposed to do on click? (for me they don't do anything)

**Comment by phuhl** [comment]
Hey,

1. yeah, buttons are defined by the sender. Take a look at the --action parameter of https://github.com/phuhl/notify-send.py/
2. Styling can be done through CSS. If you with the code from this PR https://github.com/phuhl/linux_notification_center/pull/138 you have a lot better styling options, though as you can create CSS classes that can be triggered only for certain notifications.
3. The buttons don't do anything by them-selfs. It always depends on the caller what they are supposed to do. The way it works is defined by the notification specification. Basically, when a button is clicked, the notification center sends a message to the caller via dbus and the caller can act upon that.

Hope that helps ;)
